migrate/repo_checks.py

```diff
@@ -78,6 +97,255 @@ def dry_run(self):
         raise NotImplementedError
 
 
+class EnsureWorkflowTemplates(Check):
+    """
+    There are certain github action workflows that we to exist on all
+    repos exactly as they are defined in the `.github` repo in the org.
+
+    Check to see if they're in a repo and if not, make a pull request
+    to add them to the repository.
+    """
+
+    def __init__(self, api: GhApi, org: str, repo: str):
+        super().__init__(api, org, repo)
+
+        # Heads up: This repo check currently does not currently handle the $default-branch
+        # placeholder. If you want to add a template here that uses $default-branch,
+        # then you'll need to first make the repo check aware of $default-branch.
+        self.workflow_templates = [
```

Comment:
[future work question] how might we enforce workflows on some repos (like update_python_requirements)?

Reply:
There are a few options, one straightforward one would be that some workflows have their own check class.  In the case of python requirements for example, this would mean we would have an `is_relevant` function that would would check to see if there are requirements files and maybe even that there is an `upgrade` make target.  In those cases it would ensure that the workflow exists and is up to date.

This will also require us to handle templated workflows as Kyle mentioned in his review which will probably add its own complications.